Fix typings

benny-medflyt opened the pull request:
This is a more accurate model the parse results.

See also:
<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pg-connection-string/index.d.ts>

hjr3 commented on index.d.ts:
```diff
@@ -2,13 +2,13 @@ export function parse(connectionString: string): ConnectionOptions;
+  password?: string;
+  user?: string;
+  port: string | null;
+  database: string | null | undefined;
```
An optional parameter essentially means `| undefined`. Should be `database?: string | null;`

hjr3 commented on index.d.ts:
```diff
@@ -2,13 +2,13 @@ export function parse(connectionString: string): ConnectionOptions;
+  user?: string;
+  port: string | null;
+  database: string | null | undefined;
+  client_encoding?: string | undefined;
```
The extra `| undefined` is redundant.

benny-medflyt commented:
I've updated the pull request.

I've made a few more fields optional, here is an example where only `host` and `database` are returned:

```
> p.parse("/")
{ host: '/', database: undefined }
```

I also fixed the type of `ssl` (it can be a string).

I left the `database` field to not contain a `?` annotation, because there is a difference in TypeScript between a field being optional, vs a field that is required but may have the value "undefined"

hjr3 commented:
Thank you for following up @benny-medflyt .
